packages/dicomImageLoader/src/imageLoader/wadors/loadImage.ts

```diff
@@ -90,68 +107,114 @@ function loadImage(
 
   const start = new Date().getTime();
 
-  const promise = new Promise<DICOMLoaderIImage>((resolve, reject) => {
-    // TODO: load bulk data items that we might need
-
-    // Uncomment this on to test jpegls codec in OHIF
-    // const mediaType = 'multipart/related; type="image/x-jls"';
-    // const mediaType = 'multipart/related; type="application/octet-stream"; transfer-syntax="image/x-jls"';
-    const mediaType =
-      'multipart/related; type=application/octet-stream; transfer-syntax=*';
-    // const mediaType =
-    //   'multipart/related; type="image/jpeg"; transfer-syntax=1.2.840.10008.1.2.4.50';
-
-    function sendXHR(imageURI: string, imageId: string, mediaType: string) {
+  const { retrieveTypeId, transferSyntaxUid } = options;
+  const loaderOptions = getOptions();
+  let retrieveOptions =
+    loaderOptions.getRetrieveOptions(transferSyntaxUid, retrieveTypeId) || {};
+  const uncompressedIterator = new ProgressiveIterator<DICOMLoaderIImage>(
```

Comment:
I have a good understanding of the purpose of the progressiveIterator. It serves as the equivalent of our RequestPoolManager, am I correct?

Reply:
Sort of, it is like a promise, with a final result, but that allows earlier results to also be retrieved.